### PR TITLE
Bug 1153956 - Persist the selected revision in the url on perfherder

### PR DESCRIPTION
**Pull request by akhileshpillai**
Please have a look.

Addition of a selected Revision to URL is easier now, however the removal of selected revisions from URL is not ideal.

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/621)

### CODE REVIEW

File: ui/js/graphs.js [hunk 1]
```diff
@@ -396,6 +396,8 @@ perf.controller('GraphsCtrl', [
             $scope.selectedDataPoint = getSeriesDataPoint(item);
             showTooltip($scope.selectedDataPoint);
             updateSelectedItem();
+            if ($.inArray($scope.tooltipContent.revision, $scope.highlightedRevisions) < 0) $scope.highlightedRevisions.push($scope.tooltipContent.revision);
```
Comment (tojonmz) [inline ui/js/graphs.js]: Thanks @akhileshpillai!

I think this `if` is missing it's `{` block statement?

The line is also 157 characters long. We'd like to wrap it at ~80-100 or so.

Comment (tojonmz) [inline ui/js/graphs.js]: Oh, nevermind, I see @wlach is back, you can see his comments in the bug :)

**Comment by wlach** [comment]
Hi Akhil, good start! There are some edge cases where your patch does not work which we'll need to fix before this goes in:

* If I select an item while zoomed into a particular area of the graph, the tooltip does not come up (I think because we process the zoom directive after selecting the item)
* If I change the time range of points selected, zoom in on a specific area of the graph, or in fact just click on the graph to dismiss the tooltip, we should no longer persist the tooltip state (corresponding to the fact that the tooltip has visually disappeared)

---

Review status: 0 of 2 files reviewed, 3 unresolved discussions, all commit checks successful.

---

<sup>**[ui/js/graphs.js, line 456 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/621#-JrZNSl81ka-8IZgc_jI)** ([raw file](https://github.com/mozilla/treeherder/blob/e61c917af9fd40300c8ba150c08f90e8767fb40d/ui/js/graphs.js#L456)):</sup>
I prefer to reference object properties using the ".xxx" syntax as opposed to ["xxx"]. i.e. use $scope.selectedDataPoint.projectName as opposed to $scope.selectedDataPoint["projectName"]. Also, you're missing a space between the property and the '+' here.

---

<sup>**[ui/js/graphs.js, line 600 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/621#-JrZNk9S1aU1glygw3Rf)** ([raw file](https://github.com/mozilla/treeherder/blob/e61c917af9fd40300c8ba150c08f90e8767fb40d/ui/js/graphs.js#L600)):</sup>
No need to put object properties in quotes here.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/621)

**Comment by wlach** [comment]
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

<sup>**[ui/js/graphs.js, line 651 \[r3\]](https://reviewable.io:443/reviews/mozilla/treeherder/621#-Jrx4nIcbsDMI7jBXsfy)** ([raw file](https://github.com/mozilla/treeherder/blob/e7014bf423049d1f9b9ee6080c0cfec33c71c96c/ui/js/graphs.js#L651)):</sup>
You don't need the quotes for the object properties. 

Also, I'd just call the object "tooltip", not "tooltipObject". Sort of implied.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/621)

File: ui/js/graphs.js [hunk 2]
```diff
@@ -243,10 +243,23 @@ perf.controller('GraphsCtrl', [
...
-        deselectDataPoint();
+      $("#overview-plot").bind("plotunselected",function() {
+        $scope.zoom = {};
+        $scope.selectedDataPoint = null;
         hideTooltip();
+        updateDocument();
+        // refetch and re-render all graph data
+        $q.all($scope.seriesList.map(getSeriesData)).then(function() {
+          plotGraph();
+        });
```
Comment (wlach) [inline ui/js/graphs.js]: You don't need to refetch the data here, I don't think, just re-render it. So remove the q.all() promise, and inline the plotGraph() -- probably something like:

    updateDocument()
    plotGraph(); // re-render all graph data

**Comment by wlach** [comment]
This is very close: I tested it and it works well. One minor comment. Also, the tooltip stuff is included in this PR -- we don't want that. :) Could you rebase it out? Just dropping the commit on this branch should work. Not sure if you've done that before, google/yahoo 'git rebase' if you haven't.

**Comment by edmorley** [comment]
Closing in favour of #951.

**Pull request closed without merging**